integrations/amazon_sagemaker/src/haystack_integrations/components/generators/amazon_sagemaker/sagemaker.py

```diff
@@ -0,0 +1,199 @@
+import json
+import logging
+import os
+from typing import Any, ClassVar, Dict, List, Optional
+
+import requests
+from haystack import component
+from haystack.lazy_imports import LazyImport
+from haystack_integrations.components.generators.amazon_sagemaker.errors import (
+    AWSConfigurationError,
+    SagemakerInferenceError,
+    SagemakerNotReadyError,
+)
+
+with LazyImport(message="Run 'pip install boto3'") as boto3_import:
+    import boto3  # type: ignore
+    from botocore.client import BaseClient  # type: ignore
+
+
+logger = logging.getLogger(__name__)
+
+
+MODEL_NOT_READY_STATUS_CODE = 429
+
+
+@component
+class SagemakerGenerator:
+    """
+    Enables text generation using Sagemaker. It supports Large Language Models (LLMs) hosted and deployed on a SageMaker
+    Inference Endpoint. For guidance on how to deploy a model to SageMaker, refer to the
+    [SageMaker JumpStart foundation models documentation](https://docs.aws.amazon.com/sagemaker/latest/dg/jumpstart-foundation-models-use.html).
+
```

Comment:
Let's leave a note somewhere here that SageMaker doesn't support streaming (at least they didn't 6 months ago). Please check

Reply:
Opened an issue for this feature, let's do it in the next iteration so at least we get basic support out of the door quickly https://github.com/deepset-ai/haystack-core-integrations/issues/285